game-backend/src/game-engine/mechanics/powerup-spawner/powerups/teleport.test.ts

```diff
@@ -0,0 +1,42 @@
+import { State, GameObjectTypes, UserContext } from '@darwin/types';
+import StateBuilder from '../../../../test-helper/StateBuilder';
+import { getUnit } from '../../../../helper/gameObjects';
+import consumeTeleport from './teleport';
+import { Position } from '../../../../../../darwin-types/dist/Position';
+
+describe('teleport powerup consumption', () => {
+  const NORMAL_UNIT_ID = 'unit';
+  const POWERUP_ID = 'teleport';
+
+  const normalUserContext: UserContext = {
+    unitId: NORMAL_UNIT_ID,
+  };
+
+  const initialPosition: Position = {
+    x: 1,
+    y: 1,
+  };
+
+  const state: State = StateBuilder.buildState()
+    .addUnit({
+      id: NORMAL_UNIT_ID,
+      x: 1,
```

Comment:
If you declare the initialPosition as a separate variable then i would suggest to use it here:
`addUnit({ id: NORMAL_UNIT_ID, ...position })`
Also omit the empty attributes object as it is not required.